wifiphisher/deauth.py

```diff
@@ -0,0 +1,161 @@
+
+        # broadcast address that all devices would respond to
+        broadcast_address = "ff:ff:ff:ff:ff:ff"
+
+        # check if the packet has a dot11 layer
+        if packet.haslayer(dot11.Dot11):
+            # check if the packet has either control = 0, management = 1
+            # or data = 2 as its type
+            if packet.type in [0, 1, 2]:
+                # check if client is new and connected to target access point
+                if (packet.addr1 == self._ap_bssid and
+                        packet.addr2 not in self._observed_clients and
+                        packet.addr2 is not None):
+                    # add new client to the list
+                    print packet.addr1
+                    print packet.addr2
+                    print
+                    self._observed_clients.append(packet.addr2)
+                elif (packet.addr2 == self._ap_bssid and
+                      packet.addr1 != broadcast_address and
+                      packet.addr1 not in self._observed_clients):
+                    # add new client to the list
+                    self._observed_clients.append(packet.addr1)
+
+    def find_clients(self):
+        """
+        Find all the clients
+
+        :param self: A Deauthentication object.
+        :type self: Deauthentication
+        :return: None
+        :rtype: None
+        """
+
+        # continue to find clients until otherwise told
+        while self._should_continue:
+            dot11.sniff(iface=self._jamming_interface, prn=self.process_packet,
+                        count=1, store=0)
+
+    def get_clients(self):
+        """
+        Get all the observed clients.
+
+        :param self: A Deauthentication object.
+        :type self: Deauthentication
+        :return: A list of all the observed clients.
+        :rtype: list
+        """
+
+        return self._observed_clients
+
+    def stop_deauthentication(self):
+        """
+        Stop the deauthentication process.
+        """
+
+        self._should_continue = False
+
+    def send_deauthentication_packets(self):
+        """
+        Send deauthentication packets using RadioTap header.
+
+        :param self: A Deauthentication object.
+        :type self: Deauthentication
+        :return: None
+        :rtype: None
+        .. note: Information regarding IEEE 802.11 and for deauthentication
+                 which could be useful for maintenance purposes. Type could
+                 have values of 0 for managment, 1 for control, 2 for data.
+                 There are a lot of subtpyes but subtype 12 is for
+                 deauthentication packets. addr1, addr2, addr3 are destination
+                 address, sender address, sender transmited address
+                 respectivly.
+
+
+        """
+
+        # continue to deauthenticate until otherwise set
+        while self._should_continue:
+            # added to reduce the stress on system and allow user to connect
+            time.sleep(3)
```

Comment:
Again, what kind of stress do you have in mind? We need to be as fast as possible.